sympy/polys/rings.py

```diff
@@ -1227,76 +986,9 @@ def __pow__(self, n):
             else:
                 raise ValueError("0**0")
         elif len(self) == 1:
-            monom, coeff = list(self.items())[0]
-            p = ring.zero
-            if coeff == ring.domain.one:
-                p[ring.monomial_pow(monom, n)] = coeff
-            else:
-                p[ring.monomial_pow(monom, n)] = coeff**n
-            # p._check()
-            return p
-
-        # For ring series, we need negative and rational exponent support only
-        # with monomials.
-        n = int(n)
-        if n < 0:
-            raise ValueError("Negative exponent")
-
-        elif n == 1:
-            return self.copy()
-        elif n == 2:
-            return self.square()
-        elif n == 3:
-            return self*self.square()
-        elif len(self) <= 5: # TODO: use an actual density measure
-            return self._pow_multinomial(n)
-        else:
-            return self._pow_generic(n)
-
-    def _pow_generic(self, n):
-        p = self.ring.one
-        c = self
-
-        while True:
-            if n & 1:
-                p = p*c
-                n -= 1
-                if not n:
-                    break
-
-            c = c.square()
-            n = n // 2
-
-        return p
-
-    def _pow_multinomial(self, n):
-        multinomials = multinomial_coefficients(len(self), n).items()
-        monomial_mulpow = self.ring.monomial_mulpow
-        zero_monom = self.ring.zero_monom
-        terms = self.items()
-        zero = self.ring.domain.zero
-        poly = self.ring.zero
-
-        for multinomial, multinomial_coeff in multinomials:
-            product_monom = zero_monom
-            product_coeff = multinomial_coeff
-
-            for exp, (monom, coeff) in zip(multinomial, terms):
-                if exp:
-                    product_monom = monomial_mulpow(product_monom, monom, exp)
-                    product_coeff *= coeff**exp
-
-            monom = tuple(product_monom)
-            coeff = product_coeff
-
-            coeff = poly.get(monom, zero) + coeff
-
-            if coeff:
-                poly[monom] = coeff
-            elif monom in poly:
-                del poly[monom]
+            return self._pow_single_term(n)
 
-        return poly
+        return self._pow_int(n)
```

Comment:
I don't think we need to handle special cases here like `_pow_single_term`. It can just be `_pow_int` with the clear intention that `n` should be a nonnegative integer and then `_pow_int` can handle special cases. When using python-flint I don't think there is a need to handle special cases here.